### PR TITLE
MailWatch namespace for all files (includes #843)

### PR DESCRIPTION
**Pull request by Skywalker-11**
Based on #843 and additionally introduce a MailWatch namespace :smile: 
Not sure if we need that now. But now that we have it...

### CODE REVIEW

File: tools/Cron_jobs/mailwatch_db_clean.php [hunk 1]
```diff
@@ -28,7 +28,7 @@
 namespace MailWatch;
 
 // Edit if you changed webapp directory from default
-$pathToFunctions = '/var/www/html/mailscanner/functions.php';
+$pathToFunctions = '/opt/1.2.0/mailscanner/functions.php';
```
Comment (stefaweb) [inline tools/Cron_jobs/mailwatch_db_clean.php]: `/opt/1.2.0/mailscanner/`

???

Comment (Skywalker-11) [inline tools/Cron_jobs/mailwatch_db_clean.php]: Fixed

**Comment by Skywalker-11** [comment]
By the way it is also part of the psr-1 convention for code for php < 5.3

File: mailscanner/database.php [hunk 2]
```diff
@@ -41,11 +41,11 @@ private function __construct()
      * @param string $username
      * @param string $password
      * @param string $database
-     * @return mysqli
+     * @return \mysqli
      */
     public static function connect($host = '', $username = '', $password = '', $database = '')
     {
-        if (!self::$link instanceof mysqli) {
+        if (!self::$link instanceof \mysqli) {
```
Comment (stefaweb) [inline mailscanner/database.php]: For my knowledge, what does this `\` means?

Comment (Skywalker-11) [inline mailscanner/database.php]: It means that the class is in the root namespace which is the default if you are not setting a namespace in a php file. It is also the namespace where the standard php classes are. If you have a namespace set it is assumed that all classses you are using are in the mentioned namespace and if.you want to use classes in.the root namespace you have to prefix that classes with a `\`. You can also use `use ... as ...` to set a namespace for a single class in the current file

Comment (stefaweb) [inline mailscanner/database.php]: Thanks.

**Comment by endelwar** [comment]
Well namespaces are good, but it's a bit to early ;)
I think that a modernizing refactory is what should be done for MailWatch or we will never move on to 2.0
A quick roadmap (a new post will be prepared in the next days) could be this:
- fix standing 1.2 bugs (~40 issue not tagged 2.0) ETA End of August?
- start refactoring in a different branch
- implement composer and autoloading
- consolidate all functions in separated and organized classes
- remove all globals (dependency injection) and create factories classes
- write tests!!!
- replace pear libs with something better
- make better cli script
- split domain logic from view
- implement request/response login
- reorganize directory structure (/opt/mailwatch)
- implement a router
- new UI layout

**Comment by stefaweb** [comment]
Can we publish 1.2.4 before?

(When the last cosmetic changes will be over in report graphs)

**Comment by asuweb** [comment]
@stefaweb - see the latest milestones https://github.com/mailwatch/MailWatch/milestones

**Comment by Skywalker-11** [comment]
OK I will close this then

**Pull request closed without merging**